src/app/Http/Controllers/UserController.php

```diff
@@ -5,20 +5,28 @@
 use App\Models\Post;
 use App\Models\User;
 use Illuminate\View\View;
+use App\Services\UserFollowService;
 use Illuminate\Http\RedirectResponse;
 
 class UserController extends Controller
 {
+    protected UserFollowService $userFollowService;
+
+    public function __construct(UserFollowService $userFollowService)
+    {
+        $this->userFollowService = $userFollowService;
+    }
+
     public function show(): View
     {
-        $user = optional(User::find(auth()->id()));
+        $user = User::find(auth()->id());
 
         if (!$user) {
             return view('user.profile', ['posts' => collect()]);
```

Comment:
Do you really need to add this condition here? Seems kinda pointless if the allowed user to access this route is always authenticated.